projects/angular/src/forms/common/wrapped-control.ts

```diff
@@ -131,14 +128,25 @@ export class WrappedFormControl<W extends DynamicWrapper> implements OnInit, DoC
       this._id = this.controlIdService.id;
     }
 
-    if (this.ngControlService && this.ngControl) {
-      this.ngControlService.setControl(this.ngControl);
+    if (this.ngControlService && this._ngControl) {
+      if (!this.ngControlService.getControl()) {
+        this.ngControl = this._ngControl;
+        this.ngControlService.setControl(this.ngControl);
+        this.differ = this.differs.find(this._ngControl).create();
+      } else {
+        this.ngControl = this.ngControlService.getControl();
+        this.ngControlService.addAdditionalControl(this._ngControl);
+      }
     }
   }
 
   ngDoCheck() {
-    if (this.differ) {
-      const changes = this.differ.diff(this.ngControl);
+    this.triggerDoCheck(this.differ, this.ngControl);
+  }
+
+  triggerDoCheck(differ, ngControl) {
```

Comment:
Why do we need this as separate public method? I don't see it used anywhere else.

Reply:
Basically added it as a separate methods in order to support Multiple Form controls(Primary & Additional Controls). Removed Purposefully Keyvalue differ for additional Controls now its added back.

Reply:
It's only called from the ngDoCheck method above. If you want to keep it for better code structure and readability, I am okay with this, but it can be private.